### PR TITLE
Autodetect dotenv and load it if available

### PR DESCRIPTION
**Pull request by reidmix**
Autodetect dotenv and load it if available

Github diff is being silly, I moved `lib/configerator.rb` to `lib/configerator/configerator.rb`

@heroku/sre

### CODE REVIEW

**Comment by reidmix** [comment]
I also added myself to the gemspec, because :P

File: lib/configerator.rb [hunk 1]
```diff
@@ -1,71 +1,26 @@
...
+  # dotenv-rails not available
+end
 
-  def create(name, value, error_on_load=true)
-    instance_variable_set(:"@#{name}", value)
-    instance_eval "def #{name}; @#{name} || (raise \"key not set '#{name}'\" unless #{error_on_load}) end"
-    instance_eval "def #{name}?; !!@#{name} end", __FILE__, __LINE__
+begin
+  # fallback to Dotenv
+  unless defined?(Dotenv::Railtie)
```
Comment (jmervine) [inline lib/configerator.rb]: Maybe

```
begin
    require 'dotenv'
    Dotenv.load
rescue LoadError
end unless defined?(Dotenv::Railtie)
```

Comment (reidmix) [inline lib/configerator.rb]: Oh, yeah, nice.  Fixed in c2046644294ee882efd19766c59b93fd7dda5853

**Comment by joshuatobin** [comment]
@reidmix You need a rebase, but otherwise 👍

**Comment by reidmix** [comment]
rebased, also waiting on @jmervine for another review pass.

File: configerator.gemspec [hunk 2]
```diff
@@ -3,9 +3,10 @@ Gem::Specification.new do |s|
   s.version     = '0.0.4'
   s.summary     = 'Configerator: A Config Helper'
   s.description = 'Simple module for implementing environment based configuration adapted from Pliny and following the 12factor pattern.'
-  s.authors     = ['Joshua Mervine']
-  s.email       = 'joshua@mervine.net'
+  s.authors     = ['Joshua Mervine',     'Reid MacDonald']
```
Comment (jmervine) [inline configerator.gemspec:6]: Yay! Was going to tell you to do that. :D

File: configerator.gemspec [hunk 3]
```diff
@@ -3,9 +3,10 @@ Gem::Specification.new do |s|
   s.version     = '0.0.4'
```
Comment (jmervine) [inline configerator.gemspec:3]: Going to release this along with the other two PRs as `0.0.3`. You can either update it here or I'll updated it before releasing.

**Comment by jmervine** [comment]
👍 aside from my one comment RE version, which can wait until master or be fixed when I rebase this in to my PR.

**Pull request merged**